src/gurt/telemetry.c

```diff
@@ -834,12 +780,109 @@ d_tm_compute_duration_stats(struct d_tm_node_t *node)
  * faster access.
  *
  * \param[in,out]	metric	Pointer to the metric
- * \param[in]		item	Full path name to this \a item
- *				If supplied, the path name is specified by the
- *				\a item and any strings following it.  The last
- *				item must be NULL.
+ * \param[in]		fmt	Format specifier for the name and full path of
+ *				the metric followed by optional args to
+ *				populate the string, printf style.
+ *				The format specifier and optional arguments
+ *				are used only if the pointer to the metric
+ *				is NULL.
+ *
+ * \return			DER_SUCCESS		Success
+ *				-DER_EXCEEDS_PATH_LEN	The full name length is
+ *							too long
+ *				-DER_OP_NOT_PERMITTED	Operation not permitted
+ *							because the \a item is
+ *							not a counter
+ *				-DER_UNINIT		API not initialized
+ *				-DER_INVAL		\a metric and \a item
+ *							are NULL
+ */
+int
+d_tm_set_counter(struct d_tm_node_t **metric, uint64_t val,
```

Comment:
Suggestion: Instead of allowing a counter to be set to an arbitrary value, instead enforce in the API that counters should only increment. One way this might be done is to change this function from `_set_counter()` to something like `_add_to_counter()`, and then the supplied value is added to the current value. If the anticipated use case for `_set_counter()` was to allow initializing the counter to some starting value other than zero, then the end result would be the same for the first call, and would prevent accidental reset of the counter to a smaller value.

The reasoning here is that counters should never decrease (barring rollover), as many consumers may get pretty confused by counters that aren't guaranteed to always increase. Also, allowing a counter to be set to an arbitrary value raises the question of how the type is different from a gauge.

Reply:
I'll go ahead and make that change.  It will accomplish the goal that @johannlombardi had of being able to set an initial value.